sma-server/src/main/java/fr/insalyon/creatis/sma/server/SmaServer.java

```
@@ -0,0 +1,56 @@
+package fr.insalyon.creatis.sma.server;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import org.apache.log4j.Logger;
+import org.apache.log4j.PropertyConfigurator;
+
+import fr.insalyon.creatis.sma.common.Communication;
+import fr.insalyon.creatis.sma.server.execution.Executor;
+import fr.insalyon.creatis.sma.server.execution.MessageCleanerPool;
+
+/**
+ * SmaServer
+ */
+public class SmaServer extends Thread {
+
+    private static final Logger logger = Logger.getLogger(Main.class);
+
+    private boolean started = false;
+
+    public synchronized void waitToBeReady() {
+        while (started == false) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {}
```

it should not happen but you should no silence an exception.
I think it is only needed for testing, so let that exception bubble up.